README.md

```diff
@@ -1 +1,523 @@
+{
+  "set-dns": {
+    "with": "dns://example.com?TYPE=TXT",
+    "do": "crud/update",
+    "inputs": { 
+      "value": "hello world"
+    },
+    "meta": {
+      "ipvm/config": {
+        "secret": false
+        "timeout": [500, "milli", "seconds"],
+        "verification": "attestation"
+      }
+    }
+  }
+}
+```
+
+``` json
+{
+  "run-docker": {
+    "with": "ipfs://bafkreidvq3uqoxcxr44q5qhgdk5zk6jvziipyxguirqa6tkh5z5wtpesva",
+    "do": "docker/run",
+    "inputs": {
+      "entrypoint": ["/", "home"],
+      "mounts": {"mnt": "/"},
+      "outputs": {"tmp": "/tmp"},
+      "env":  {"$HELLO": "world"},
+      "workdir":  "/work"
+    },
+    "meta": {
+      "ipvm/config": {
+        "secret": false,
+        "check": {
+          "optimistic": 2, 
+          "referee": "did:key:zStEZpzSMtTt9k2vszgvCwF4fLQQSyA15W5AQ4z3AR6Bx4eFJ5crJFbuGxKmbma4"
+        }
+      }
+    }
+  }
+}
+```
+
+``` json
+{
+  "pipelined-docker-promise": {
+    "with": {"ucan/promise": ["/", "previous_action"]},
+    "do": "docker/run",
+    "inputs": {
+      "entrypoint": ["/", "home"],
+      "mounts": {"mnt": "/"},
+      "outputs": {"tmp": "/tmp"},
+      "env":  {"$HELLO": "world"},
+      "workdir":  "/work"
+    },
+    "meta": {
+      "ipvm/config": {
+        "disk": "200GB"
+      }
+    }
+  }
+}
+```
+
+# 5 Exception Handler
+
+If present, the OPTIONAL `catch` field MUST be run in response to a `Task` returning on the `Failure` branch. The deterministic & pure Wasm module MUST take a `Failure` object as input, and MUST return data in the following shape:
+
+``` ipldsch
+type Handle union {
+  | Success "ok"     -- End task with Success object
+  | String  "rewire" -- Task name inside the current Workflow
+  | String  "msg"    -- Format the error message and panic
+} respresentation keyed
+```
+
+If the `msg` branch is returned, the invocation MUST immediately rethrow with the update message.
+
+Note that while IPVM MUST treat the pure tasks together as transactional. It is not possible to roll back any destructive effects that have already been run. As such, it is RECOMMENDED to have few (if any) tasks depend on the output of a destructive effect, so they can be scheduled at the end of the workflow.
+
+# 6 Receipt Output
+
+| Field  | Type            | Description                                                           | Required | Default |
+|--------|-----------------|-----------------------------------------------------------------------|----------|---------|
+| `inv`  | `&Invocation`   | CID of the Invocation that generated this response                    | Yes      |         |
+| `out`  | `{String: Any}` | The results of each call, the task's label. MAY contain sub-receipts. | Yes      |         |
+| `meta` | `Any`           | Non-normative extended fields                                         | No       | `null`  |
+
+If the `catch` field is set on the outer `Workflow`, The `out` field MAY include the output under the `ipvm/catch` key
```

Comment:
would we have a need to actively want to `throw` an exception based on the execution of a task?

Reply:
We also covered this on the call after this was posted. For posterity:

I've been thinking about this as akin to stdout and stderr. If you run an HTTP PUT, and it fails, the runtime needs to return that as a `Failure`. If a Docker container exits with a nonzero exit code, we'll capture its stderr and put that in the `Failure` struct.

Good point: this needs to be in the text 👍